arangod/Aql/QueryContext.h

```diff
@@ -155,6 +168,8 @@ class QueryContext {
   constexpr static std::size_t baseMemoryUsage = 8192;
 
  protected:
+  std::unique_ptr<memory_resource_t> _memoryResource;
```

Comment:
Why unique_ptr needed? You could just put here CountingMemoryResourse.
Probably even better you could put ResourceMonitor to it

Reply:
I still don't understand why this PR needs pmr and why not just own allocator?